lib/std/os.zig

```diff
@@ -5021,6 +5022,89 @@ pub fn copy_file_range(fd_in: fd_t, off_in: u64, fd_out: fd_t, off_out: u64, len
     return pwrite(fd_out, buf[0..amt_read], off_out);
 }
 
+var has_copy_file_range_syscall = std.atomic.Int(u1).init(1);
+
+pub const CopyFileOptions = struct {};
+
+pub const CopyFileError = error{
+    BadFileHandle,
+    SystemResources,
+    FileTooBig,
+    InputOutput,
+    IsDir,
+    OutOfMemory,
+    NoSpaceLeft,
+    Unseekable,
+    PermissionDenied,
+    FileBusy,
+} || FStatError || SendFileError;
+
+/// Transfer all the data between two file descriptors in the most efficient way.
+/// No metadata is transferred over.
+pub fn copy_file(fd_in: fd_t, fd_out: fd_t, options: CopyFileOptions) CopyFileError!void {
```

Comment:
What is the "posix layer" API function that this represents? It looks like the only one that has such a function is Darwin, in which case this should be called `fcopyfile` and not have an options parameter.

Reply:
There's no POSIX equivalent, we can just move this into `fs` as a private helper and call it a day.

Reply:
Sounds good :+1:

Reply:
btw if you haven't seen it: #5019

Reply:
> btw if you haven't seen it: #5019

IMO we don't need a posix layer, we only need a `std.os` that _does_ things (eg. `copyFile`, `writeData`) rather than trying to paper over all the differences between different OSs. If one wants a specific posix-specific/Windows-specific function they're more than welcome to do so by explicitly writing it out, but a full-blown Posix compatibility layer is not something that belongs into a stdlib.

Reply:
I agree that creating a full blown posix compatibility layer is not a goal in and of itself. But it's an implementation detail of providing a std that does things (e.g. `std.fs`). It's code that we would end up writing - in fact did end up writing - as an implementation detail of the higher level cross platform abstractions. "posix layer" is just a way to think about how it is organized.